src/Middleware/HttpLogging/src/W3CLoggerProcessor.cs

```diff
@@ -35,75 +35,82 @@ public override async Task OnFirstWrite(StreamWriter streamWriter, CancellationT
             }
             if (_loggingFields.HasFlag(W3CLoggingFields.Method))
             {
-                await WriteField(" cs-method", streamWriter, cancellationToken);
+                await streamWriter.WriteAsync(" cs-method".AsMemory(), cancellationToken);
             }
             if (_loggingFields.HasFlag(W3CLoggingFields.UriStem))
             {
-                await WriteField(" cs-uri-stem", streamWriter, cancellationToken);
+                await streamWriter.WriteAsync(" cs-uri-stem".AsMemory(), cancellationToken);
             }
             if (_loggingFields.HasFlag(W3CLoggingFields.UriQuery))
             {
-                await WriteField(" cs-uri-query", streamWriter, cancellationToken);
+                await streamWriter.WriteAsync(" cs-uri-query".AsMemory(), cancellationToken);
             }
             if (_loggingFields.HasFlag(W3CLoggingFields.ProtocolStatus))
             {
-                await WriteField(" sc-status", streamWriter, cancellationToken);
+                await streamWriter.WriteAsync(" sc-status".AsMemory(), cancellationToken);
             }
             if (_loggingFields.HasFlag(W3CLoggingFields.TimeTaken))
             {
-                await WriteField(" time-taken", streamWriter, cancellationToken);
+                await streamWriter.WriteAsync(" time-taken".AsMemory(), cancellationToken);
             }
             if (_loggingFields.HasFlag(W3CLoggingFields.ProtocolVersion))
             {
-                await WriteField(" cs-version", streamWriter, cancellationToken);
+                await streamWriter.WriteAsync(" cs-version".AsMemory(), cancellationToken);
             }
             if (_loggingFields.HasFlag(W3CLoggingFields.Host))
             {
-                await WriteField(" cs-host", streamWriter, cancellationToken);
+                await streamWriter.WriteAsync(" cs-host".AsMemory(), cancellationToken);
             }
             if (_loggingFields.HasFlag(W3CLoggingFields.UserAgent))
             {
-                await WriteField(" cs(User-Agent)", streamWriter, cancellationToken);
+                await streamWriter.WriteAsync(" cs(User-Agent)".AsMemory(), cancellationToken);
             }
             if (_loggingFields.HasFlag(W3CLoggingFields.Cookie))
             {
-                await WriteField(" cs(Cookie)", streamWriter, cancellationToken);
+                await streamWriter.WriteAsync(" cs(Cookie)".AsMemory(), cancellationToken);
             }
             if (_loggingFields.HasFlag(W3CLoggingFields.Referer))
             {
-                await WriteField(" cs(Referer)", streamWriter, cancellationToken);
+                await streamWriter.WriteAsync(" cs(Referer)".AsMemory(), cancellationToken);
             }
+
+            await streamWriter.FlushAsync();
         }
 
         internal async Task WriteField(string messageField, StreamWriter streamWriter, CancellationToken cancellationToken)
```

Comment:
Do we still need this?

Reply:
No, I'll remove it. I was trying to write directly the fields in the streamwriter with this method, but I got a timeout with the test 🙂 . We can try a PR on that with a good refacto.